test_plans/exceptions.asciidoc

```diff
@@ -0,0 +1,246 @@
+
+* `exception::has_context()` should return true
+* `exception::get_context()` should return the ctx passed
+
+==== exception(context ctx, int env, const std::error_category& cat)
+
+Same as `exception(int env, const std::error_category& cat)`, but:
+
+* `exception::has_context()` should return true
+* `exception::get_context()` should return the ctx passed
+
+==== exception(std::error_code ec, const std::string& what_arg)
+
+Same as `exception(std::error_code)`, but:
+
+* for non-empty `what_arg` `exception::what()` should contain `what_arg` as substring
+* for empty `what_arg` `exception::what()` should return null-terminated string
+
+==== exception(std::error_code ec, const char* what_arg)
+
+Same as `exception(std::error_code ec, const std::string& what_arg)`.
+
+==== exception(int env, const std::error_category& cat, const std::string& what_arg)
+
+Same as `exception(int env, const std::error_category& cat)`, but
+same checks for `what_arg` as for
+`exception(std::error_code ec,  const std::string& what_arg)`
+
+==== exception(int env, const std::error_category& cat, const char* what_arg)
+Same as `exception(int env, const std::error_category& cat, const std::string& what_arg)`
+
+==== exception(context ctx, std::error_code ec, const std::string& what_arg)
+
+Same as `exception(context ctx, std::error_code ec)`, but:
+same checks for `what_arg` as for
+`exception(std::error_code ec,  const std::string& what_arg)`
+
+==== exception(context ctx, std::error_code ec, const char* what_arg)
+Same as `exception(context ctx, std::error_code ec, const std::string& what_arg)`
+
+==== exception(context ctx, int env, const std::error_category& cat, const std::string& what_arg)
+
+Same as `exception(context ctx, int env, const std::error_category& cat)`, but:
+same checks for `what_arg` as for
+`exception(std::error_code ec,  const std::string& what_arg)`
+
+==== exception(context ctx, int env, const std::error_category& cat, const std::string& what_arg)
+
+Same as `exception(context ctx, int env, const std::error_category& cat, const std::string& what_arg)`
+
+==== exception( const exception& other )
+
+For default constructor and any other 12 constructors
+make same checks as for the original constructor
+
+==== operator = ()
+
+* For any constructor from (default constructor and 12 other constructors) as SRC object:
+  for any second constructor from ( default and 12 custom constructors) as DST object:
+  make same checks as for the SRC constructor
+* ensure copy assignment doesn't throw
+
+=== sycl::exception_list
+
+Verify that:
+
+* `value_type` `std::is_same` with `std::exception_ptr`
+* `reference` `std::is_same` with `value_type&`
+* `const_reference` `std::is_same` with `const value_type&`
+* `size_type` `std::is_same` with `std::size_t`
+* `iterator` satisfies `LegacyForwardIterator`
+* `const_iterator` satisfies `LegacyForwardIterator`
+* `exception_list::size()` is zero
+* `++exception_list::begin()` is equal to `exception_list::end()`
```

Comment:
I presume these last two tests will be performed on a default-constructed `exception_list`?